### ISSUE TITLE
Installation issue: py-llvmlite

### ISSUE THREAD

**Issue by rgmiller** [spack/spack#9667]
py-llvmlite has an issue with the dependencies.  The relevant line in package.py is
```python
depends_on('llvm@6.0:', when='@0.23.0:')
```
Note the colon after the version number.  Spack will install anything newer than 6.0 and 7.0.0 is now available.  However, the setup.py file in py-llvmlite code insists on major version 6.  The simple solution is to just change the version in the depends_on() call to "6.0.1" (without the colon).  We might also want to check and see if a newer version of llvmlite exists that works with LLVM 7.x

### Steps to reproduce the issue

```console
$ spack install py-llvmlite
```

```
==> Error: ProcessError: Command exited with status 1:
    '/home/xmr/git/spack/opt/spack/linux-centos7-x86_64/gcc-8.2.0/python-3.7.0-s5za75wfhglupziwnnwnnjq6ek7nlwef/bin/python3.7' '-s' 'setup.py' '--no-user-cfg' 'build'

1 error found in build log:
     13        main_posix('linux', '.so')
     14      File "/tmp/xmr/spack-stage/spack-stage-hQ1bbv/llvmlite-0.23.0/ffi/build.py",
            line 105, in main_posix
     15        raise RuntimeError(msg)
     16    RuntimeError: Building llvmlite requires LLVM 6.0.x. Be sure to set LLVM_CONFI
           G to the right executable path.
     17    Read the documentation at http://llvmlite.pydata.org/ for more information abo
           ut building llvmlite.
     18    
  >> 19    error: command '/home/xmr/git/spack/opt/spack/linux-centos7-x86_64/gcc-8.2.0/p
           ython-3.7.0-s5za75wfhglupziwnnwnnjq6ek7nlwef/bin/python3.7' failed with exit s
           tatus 1
```

Note line 16: we need LLVM 6.0.x, but the version of llvm that is installed by default is 7.0.0.

### Platform and user environment

Please report your OS here:
```
xmr@boo4:~$ uname -a
Linux boo4 4.15.0-36-generic #39-Ubuntu SMP Mon Sep 24 16:19:09 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
xmr@boo4:~$ lsb_release -d
Description:    Ubuntu 18.04.1 LTS
```

**Comment by rgmiller** [comment]
Looks like there aren't any newer versions of llvmlite.

**Comment by ax3l** [comment]
yep, (imho) an anti-pattern with an open-end dependency version. thanks a lot for fixing!

**Comment by ax3l** [comment]
Thanks again for the fix in #9668!
